3.5/SSO_README.md

```diff
@@ -0,0 +1,92 @@
+
+![saml flow](./images/saml_guidance_saml_flow.png)
+
+For Cloud Pak for Data, the CDP platform will be the Service Provider and in the example Okta will be the Identity Provider.
+
+Follow these steps to set up SSO:
+
+* [Add application to the Identity Provider](#add-application-to-the-identity-provider)
+* [Update configuration on CPD](#update-configuration-on-cpd)
+* [Add external users and test](#add-external-users-and-test)
+* [Disable the default admin user](#disable-the-default-admin-user)
+
+## Add Application to the Identity Provider
+
+Follow these steps with Okta, or any SSO IDP which does not accept an input SAML export. Begin by creating a new application in the IDP and select to use SAML 2.0 as the authenticaion method.
+
+In Okta, you will be prompted for an application name which can be "Cloud Pak for Data" or more specific as desired. It's ok to retain defaults for the rest of the settings on the first page. Proceed to the **Next** panel.
+
+* Configure the SAML properties. The most important and first is the **Single Sign On URL**, this will be required by any SAML IDP. Use the route to the instance with `/auth/login/sso/callback` appended.
+
+    ```console
+    cpdhost=$(oc get route cpd -n <cpd-namespace> -o=jsonpath='{.spec.host}{"\n"}')
+    echo "https://$cpdhost/auth/login/sso/callback"
+    ```
+
+* Define a Service Provider entity id - this is also referred to as the **issuer** in the CPD documentation. If unset, it will default to `ibm_privatecloud`. In this example the Service Provider entity id will be `cp4d-partner-roks`.
+
+    If you will use userids with no special formatting (e.g. not an email address with a `@` in the middle of the string), select the Name ID format to be **Unspecified**. If you will use userids in an email address format, change the format to **EmailAddress**. You can leave the Application username as the default as an **additional attribute** will be added in the configuration to return the user identifier in the SAML assertion. The value selected here will determine the value used in configuring CPD. If using **Unspecified** the CPD `identifierFormat` should be: `urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified` and when using **EmailAddress** the `identifierFormat` should be `urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress`.
+
+* Add an additional attribute needs to be added to specify the user identity. Presently the `nameID` property in the Okta SAML assertion is not parsed, requiring this additional attribute. In the **Attribute Statements (optional)** section add an attribute with label `username`, and specify the preferred format. For the value, select either `user.email` if the email is to be used, or if a simple string like the left-hand-side of the Okta login should be used, type in an expression like: `String.substringBefore(user.login, "@")`
```

Comment:
Extra word "Add" at beginning of sentence

Reply:
fixed in https://github.com/ibm-hcbt/cp4d-install/pull/3/commits/81e270188e020b8dc1b641ee6017216a4f63c953